venia-integration-tests/src/support/visitPageCommand.js

```diff
@@ -0,0 +1,6 @@
+const visitPage = (routeUrl, wait = 5000) => {
```

Comment:
I am not convinced that this approach is going to match end user behaviour. An end user will click themselves through the react app to navigate. This has a different behaviour to typing the url in the address bar.  So each test should at max have 1 visitPage use at the very start as the entrypoint.

Additionally I suggest removing the hard coded wait. From experience I can say that some will require a lot more than 5000, depending on back-end. It would be best to wait for the appropriate spinning loader to no longer be present or the next element required for the next interaction being there.

Reply:
Yeah agree on that point @fooman however these direct page visit would be used to avoid retesting already tested user flow part of other tests which makes test more lighter, stable and decoupled, examples -
1. Add product to cart - Actual user navigation to PDP would be part of category listing test so when need to add product to cart we would direct hit pdp.
2. Order placement flows - we will use user navigation to `cart` `checkout` pages in 1 or 2 tests but when not needed to check link functionality `Edit Shipping Bag`  `checkout` or `go to checkout` but still need to test something on checkout flows then will hit direct checkout.
3. Actual user category visits would be part of Mega menu tests but rest of the times we will directly go to needed category.

Regrading 5 sec wait - currently we dont have better way to handle full page load so might be temporary until we find better solution. One example - When we visit home page and try to create account >> cypress automatically waits and clicks on `Sign In`  button but at this time the page is still loading. Now when page loads the signin dropdown is collapsed, cypress does not find `Create an Account` button and fail the test.  In this case we did try to wait for home page banner image to load but its not same banner image always, it keeps changing and test fails randomly.